app/scenes/account_notifications/account_notifications.js

```diff
@@ -0,0 +1,515 @@
+// Copyright (c) 2017 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React, {PropTypes, PureComponent} from 'react';
+import {
+    ScrollView,
+    StyleSheet,
+    View
+} from 'react-native';
+
+import TextInputWithLocalizedPlaceholder from 'app/components/text_input_with_localized_placeholder';
+import {changeOpacity, makeStyleSheetFromTheme} from 'app/utils/theme';
+import EventEmitter from 'service/utils/event_emitter';
+import {Preferences, RequestStatus} from 'service/constants';
+import {getPreferencesByCategory} from 'service/utils/preference_utils';
+
+import Section from './section';
+import SectionItem from './section_item';
+import SaveNotifcationsButton from './save_notifications_button';
+
+const getStyleSheet = makeStyleSheetFromTheme((theme) => {
+    return StyleSheet.create({
+        input: {
+            color: theme.centerChannelColor,
+            fontSize: 12,
+            height: 40
+        },
+        separator: {
+            height: 1,
+            flex: 1,
+            backgroundColor: changeOpacity(theme.centerChannelColor, 0.1),
+            marginHorizontal: 15
+        },
+        scrollView: {
+            flex: 1,
+            backgroundColor: changeOpacity(theme.centerChannelColor, 0.03)
+        },
+        scrollViewContent: {
+            paddingBottom: 30
+        },
+        wrapper: {
+            flex: 1,
+            backgroundColor: theme.centerChannelBg
+        }
+    });
+});
+
+export default class AccountNofications extends PureComponent {
+    static propTypes = {
+        actions: PropTypes.shape({
+            goBack: PropTypes.func.isRequired,
+            handleUpdateUserNotifyProps: PropTypes.func.isRequired
+        }),
+        config: PropTypes.object.isRequired,
+        currentUser: PropTypes.object.isRequired,
+        myPreferences: PropTypes.object.isRequired,
+        saveRequestStatus: PropTypes.string.isRequired,
+        subscribeToHeaderEvent: PropTypes.func.isRequired,
+        theme: PropTypes.object.isRequired,
+        unsubscribeFromHeaderEvent: PropTypes.func.isRequired
+    };
+
+    static navigationProps = {
+        renderRightComponent: (props, emitter) => {
+            return <SaveNotifcationsButton emitter={emitter}/>;
+        }
+    }
+
+    constructor(props) {
+        super(props);
+
+        const {currentUser} = props;
+        const notifyProps = currentUser.notify_props || {};
+        this.setStateFromNotifyProps(notifyProps);
+    }
+
+    componentWillMount() {
+        this.props.subscribeToHeaderEvent('save_notify_props', this.saveUserNotifyProps);
+    }
+
+    componentWillUnmount() {
+        this.props.unsubscribeFromHeaderEvent('save_notify_props');
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (nextProps.currentUser !== this.props.currentUser) {
```

Comment:
Yes, when the ```updateUserNotifyProps``` action receives data back, it's the complete user object. The ```UserTypes.RECEIVED_ME``` event is dispatched with that user object which replaces the entire user object for that profile id. https://github.com/mattermost/mattermost-mobile/blob/master/service/reducers/entities/users.js#L96. We could check notifyProps, but since we are changing the entire user object this accomplishes the same thing.